includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -790,21 +790,14 @@ private function is_product_supported() {
 		$product      = $this->get_product();
 		$is_supported = true;
 
-		if ( is_null( $product ) ) {
-			$is_supported = false;
-		} elseif ( ! is_object( $product ) || ! in_array( $product->get_type(), $this->supported_product_types(), true ) ) {
-			$is_supported = false;
-		} elseif ( class_exists( 'WC_Subscriptions_Product' ) && $product->needs_shipping() && WC_Subscriptions_Product::get_trial_length( $product ) > 0 ) {
-			// Trial subscriptions with shipping are not supported.
-			$is_supported = false;
-		} elseif ( class_exists( 'WC_Pre_Orders_Product' ) && WC_Pre_Orders_Product::product_is_charged_upon_release( $product ) ) {
-			// Pre Orders charge upon release not supported.
-			$is_supported = false;
-		} elseif ( class_exists( 'WC_Composite_Products' ) && $product->is_type( 'composite' ) ) {
-			// Composite products are not supported on the product page.
-			$is_supported = false;
-		} elseif ( class_exists( 'WC_Mix_and_Match' ) && $product->is_type( 'mix-and-match' ) ) {
-			// Mix and match products are not supported on the product page.
+		if ( ! is_object( $product )
```

Comment:
Thanks for accommodating my suggested change :) It looks quite clean now 😄 

Should we restore the clarifying code comments from previews if conditions?

Reply:
That makes sense, I will bring back the inline comments.